### PR TITLE
AI hero smooth movement

### LINKED ISSUE #903: The screen doesn't move smoothly during AI hero turn
Notice that during AI hero move the screen doesn't smoothly move like it is done for human's hero turn.

### PR DESCRIPTION
**Pull request by idshibanov**
Fixes #903 . Optimized the code a bit, as well as fixing existing bugs with AI heroes moving from or into fog.

### CODE REVIEW

File: src/fheroes2/ai/ai_hero_action.cpp [hunk 1]
```diff
@@ -1934,19 +1939,9 @@ namespace AI
...
-                    cursor.Show();
-                    display.Flip();
                 }
             }
 
-            bool hide_move = ( 0 == conf.AIMoveSpeed() ) || ( !IS_DEVEL() && !AIHeroesShowAnimation( hero ) );
-
-            // 0.2 sec delay for show enemy hero position
-            if ( !hero.isFreeman() && !hide_move )
-                DELAY( 200 );
```
Comment (idshibanov) [inline src/fheroes2/ai/ai_hero_action.cpp:1948]: I believe with smooth movement we don't need additional delay

**Comment by ihhub** [comment]
Hi, @idshibanov , could you please test this save? Just load and end the turn. You could see that during movement of yellow player the screen moves for a hidden hero.
[Broken_Alliance_0032.zip](https://github.com/ihhub/fheroes2/files/4878912/Broken_Alliance_0032.zip)

**Comment by idshibanov** [comment]
> 
> 
> Hi, @idshibanov , could you please test this save? Just load and end the turn. You could see that during movement of yellow player the screen moves for a hidden hero.
> [Broken_Alliance_0032.zip](https://github.com/ihhub/fheroes2/files/4878912/Broken_Alliance_0032.zip)

Oversight as usual. Got it, should be fixed!

**Comment by ihhub** [comment]
Hi @idshibanov and @shprotru , thanks a lot for fixing this issue!

**Pull request merged**